README.md

```diff
@@ -1,2 +1,125 @@
-# api
+# 🚀 api
 
+[![npm](https://img.shields.io/npm/v/api)](https://npm.im/api) [![Build](https://github.com/readmeio/api/workflows/CI/badge.svg)](https://github.com/readmeio/api)
+
+Automatic SDK generation from an OpenAPI definition.
+
+* [Installation](#installation)
+* [Usage](#usage)
+    * [Authentication](#authentication)
+    * [Parameters and Payloads](#parameters-and-payloads)
+    * [HTTP requests](#http-requests)
+* [How does it work?](#how-does-it-work)
+* [Interested in contributing?](#interested-in-contributing)
+* [FAQ](#faq)
+
+## Installation
+```
+npm install api --save
+```
+
+## Usage
+Using `api` is as simple as supplying it an OpenAPI and using the SDK as you would any other!
+
+```js
+const sdk = require('api')('https://raw.githubusercontent.com/readmeio/oas/master/packages/examples/3.0/json/petstore.json');
+
+sdk.listPets().then(res => res.json()).then(res => {
+  console.log(`My pets name is ${res[0].name}!`);
+});
+```
+
+The OpenAPI definition is automatically downloaded, cached, and transformed into a chainable [`fetch`](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) Promise that you can use to make API requests.
+
+### Authentication
+`api` supports API authentication through an `.auth()` method that you can chain to your requests, as such:
+
+```js
+sdk.auth('myApiToken').listPets().then(...);
+```
+
+With the exception of OpenID, it supports all forms of authentication supported by the OpenAPI specification! Just give `.auth()` your credentials and it'll figure out how to use it according to the API you're using.
+
+For example:
+
+* HTTP Basic auth: `sdk.auth('username', 'password')`
+* Bearer tokens (HTTP or OAuth 2): `sdk.auth('myBearerToken')`
+* API Keys: `sdk.auth('myApiKey')`
+
+### Parameters and Payloads
+When supplying parameters and/or request body payloads to an API request, you don't to explicitly define what goes where since the API definition contains all that information. All you need to do is supply either one or two objects:
+
+* `body`: This will contain all data required for a request body payload for a POST, PUT, etc. request. It can either be an array or an object, whichever you need to use the API operation you're using.
```

Comment:
```suggestion
* `body`: This will contain all data required for a request body payload for a POST, PUT, etc. request. It can either be an array or an object — whichever you need to use the API operation you're using.
```